### DISCUSSION TITLE
2 Noob questions: 1. what provider is used by w3mProvider? 2. how can I query public blockchain data before connecting?

### DISCUSSION THREAD

**Discussion by dataexcess** [WalletConnect/walletconnect-monorepo#3434] [Developer Support] [answered]
Hello!

I am using web3modal to connect my user and using wagmi to interact with the blockchain. My connection looks sometthin like this (in nuxt 3):

import { UseWagmiPlugin, configureChains, createConfig, mainnet, sepolia } from 'use-wagmi'
import { publicProvider } from 'use-wagmi/providers/public'
import { EthereumClient, w3mConnectors, w3mProvider } from '@web3modal/ethereum'
import { Web3Modal } from '@web3modal/html'

```
export default defineNuxtPlugin((nuxtApp) => {
    const config = useRuntimeConfig()
    const projectId = config.public.WALLETCONNECT_PROJECT_ID
    const chains = [mainnet, sepolia]

    const { publicClient } = configureChains(chains,
        [w3mProvider({ projectId }), publicProvider()]
    )

    const web3config = createConfig({
        autoConnect: true,
        connectors: w3mConnectors({ projectId, chains }),
        publicClient
    })

    let web3Modal = null
    if (process.client) {
        const ethereumClient = new EthereumClient(web3config, chains)
        web3Modal = new Web3Modal({ projectId }, ethereumClient)
    }

    nuxtApp.vueApp.use(UseWagmiPlugin, web3config)

    return {
        provide: {
            web3Modal
        }
    }
})
```

Interacting with the blockchain works, but only after I connect the user with his wallet. I want to query some blockchain smart contract data before the user connects already however. I have the feeling that the provider is only enabled after the user connects with the modal to his/her wallet. Is this correct? Is there a fallback provider that can be used before a connection? I simply want to query all the tokenURI of my NFT contract tokens, even before the user has connected...

And two: can I use INFURA with the Web3Modal? What provider is it using now?

Sorry, these questions might be very noobish. 

Thank you!

**Comment by aliburzhi** [comment]
about INFURA
here is the docs link 
[Infura](https://docs.walletconnect.com/2.0/advanced/migration-from-v1.x/dapps#ethereum-provider)
[infura provider docs](https://docs.walletconnect.com/2.0/web3modal/react/wagmi/custom-chains#custom-chain-providers)
The new implementation requires projectId and chains instead. Here is an example of the new initialization:


```
import { EthereumProvider } from '@walletconnect/ethereum-provider'

const provider = await EthereumProvider.init({
  projectId: 'WALLETCONNECT_PROJECT_ID', // required
  chains: [1], // required
  showQrModal: true // requires @walletconnect/modal
})
```

**Comment by glitch-txs** [comment, marked as answer]
Hi! 
1. You can check the source code [here](https://github.com/WalletConnect/web3modal/blob/2a0e946f9c848c25d6c9cf3c53ab4a04ed372345/chains/ethereum/src/utils.ts#L35). It's basically using WalletConnectConnector and InjectedConnector

2. I would suggest going through the Wagmi Core documentation, you have there functions to query blockchain and user's data. About providers configuration you can check [here](https://wagmi.sh/core/providers/infura) and the second link shared above :)
